x-pack/solutions/security/plugins/security_solution/common/detection_engine/prebuilt_rules/diff/convert_rule_to_diffable.ts

```diff
@@ -224,7 +225,7 @@ const extractDiffableThresholdFieldsFromRuleObject = (
     type: rule.type,
     kql_query: extractRuleKqlQuery(rule.query, rule.language, rule.filters, rule.saved_id),
     data_source: extractRuleDataSource(rule.index, rule.data_view_id),
-    threshold: rule.threshold,
+    threshold: extractRuleThreshold(rule),
```

Comment:
What was the issue with `threshold` field?

Reply:
We default to an empty array to `cardinality` within the `threshold` field object when we save from the create/edit rule UI form resulting in weird results like this 
![image](https://github.com/user-attachments/assets/1566f9b6-92f7-448f-b41c-89e7fab9727c)
This basically normalizes that, similar to what we do in the `threat` extraction